llama_index/vector_stores/pinecone.py

```diff
@@ -210,21 +213,30 @@ def __init__(
     ) -> None:
         """Initialize params."""
         try:
-            import pinecone
+            # import pinecone
+            pinecone = __import__("pinecone")
```

Comment:
Interesting - what is the purpose behind this change? 

I read into reasons why you might want to do this - looks like dynamic importing, or handing potential import errors can be improved by this format, but it's not clear on first glance why we're doing it this way. 

I'd recommend adding some comments for context around this change.

Reply:
Will do!

(And yes, this is what ChatGPT helped me tweak so that in our unit tests, we could inject diff versions)

Reply:
done